source/common/common/mutex_tracer.cc

```diff
@@ -0,0 +1,39 @@
+#include "common/common/mutex_tracer.h"
+
+#include <iostream>
+#include <memory>
+
+#include "common/common/assert.h"
+
+#include "absl/synchronization/mutex.h"
+
+namespace Envoy {
+
+MutexTracer* MutexTracer::singleton_ = nullptr;
+
+MutexTracer* MutexTracer::getOrCreateTracer() {
+  if (singleton_ == nullptr) {
```

Comment:
We need singleton_ as a static class member to avoid the barrier-lookup (atomic or mutex or whatever is under the hood) for the tracer object on every contention.

I think adding a comment here as to why a member-var is desired makes sense.

Reply:
OK that makes sense.